### ISSUE TITLE
Lifefram - Encyclopedia of the Departed

### ISSUE THREAD

**Issue by melwong** [filecoin-project/devgrants#604]
# Open Grant Proposal: `Lifefram - Encyclopedia of the Departed`

**Name of Project: Lifefram**

**Proposal Category: app-dev**

**Proposer: melwong**

**(Optional) Technical Sponsor: Jenks Guo / https://github.com/jenks-guo-filecoin** 

**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?: Yes** 

# Project Description

Here's a quick introduction video - https://youtu.be/dbHIJiGrWEI

Around 160,000 people pass away every day in this world and yet there's no sustainable way for us to remember them. Most online memorials do not store their data beyond our lifetime and Wikipedia is only for notable people.

Which gave us the idea to build a global database to keep records of all the people who have died, permanently and persistently for 300 years and more using IPFS and Filecoin. 

This non-profit project is proud to be the grand champion of the Babylon Project 2.0 hackathon. 

We're the recipient of your microgrant and have completed all our milestones for that grant in 20 days ahead of schedule. Since our project has a global purpose of documenting individuals who have passed away around the world, it takes considerable resources to make this project a success. While we are trying to seek donations from other external parties, we are hoping Filecoin could consider providing us more financial support.

## Value

This project echoes Filecoin's mission statement of storing humanity's most important information. In our case, it's information about all of us. It has immense value to humanity where it aims to document every living person who lived in this world forever, thus, giving us a sense of immortality.

It could help our children and our children's children to learn about their ancestral roots and also assist us to predict the future by learning about our past. Case in point would be preparing for future virus outbreak where they could study those who died from past pandemics. 

Ultimately, since most of us want to be remembered one way or another, this project aims to provide the means for that.

**Risk Analysis and Mitigation**

If we don't get this project right, it will not cause any harm to anyone since each biography needs to be validated by a death certificate or comes with an attestation to the death. We will continue to operate this project.

To ensure the longevity of this project, we are constantly looking for young team members to eventually handover this project to in the future, thus making this project truly decentralized.

The areas that would make this project difficult to execute are,
* Documenting every single person who passed away in every corner of the world in different languages.
* Continuously receiving funds into the farthest future
* Continuously getting future stakeholders to be involved
* Continuously evolve with new technology trend in the far future, especially in data storage solutions

## Deliverables

* Published over 500,000 biographies of the departed globally.
* Able to archive death records of a government department.
* Able to publish and store biographies at an average cost of $0.09 (or lower) per bio. This cost includes software subscription fee, hosting fee, storage fee, labor fee, etc.
* Able to optimize and scale our biography publication and decentralized storage process. 
* Able to search for a biography within 5 seconds using our new and optimized search function.
* Improved UI/UX on our platform.

## Development Roadmap

NOTES:
* If this proposal is approved, we assume to commence work on ~~June~~ August 15, 2022. A one-year period is measured from ~~June~~ August 15, 2022 to ~~June~~ August 14, 2023.
* Manday is defined to be 8 hours of work performed by a person in a day. Only total mandays are depicted below, not continuous total days in succession. This is because some team members are working on a part-time/freelance basis. 
* We're taking the average American middle-level developer's [hourly rate of USD47](https://devoxsoftware.com/blog/average-software-developer-hourly-rate/) as a benchmark multiply by 8 hour work day to derive the cost for one manday.
* Manday work covers development, testing, staging and deployment to production.
* Content editor's proposed hourly rate is $3 per hour (minimum set by Upwork). We shall ensure we meet minimum wage requirement per country when we are hiring content curators and editors globally.

**Milestone: Improve UI/UX Of Our Platform**
* Our current design has flaws that we need to improve such as pagination issue, search function, issue on listing and biography display page.
* We also require a more robust donation page for us to collect funding that could sustain our global operation.
* Estimated mandays: 7
* Estimated number of worker: 1 (developer) 
* Funding required: $2,632 ( ($47 per hour * 8 hours ) * 7 mandays )
* Estimated date of completion: ~~July~~ Sept 15, 2022

**Milestone: Optimized Search Function**
* A site-wide search feature that allow users to quickly search for biographies across our large database.
* User shall be able to quickly and conveniently search based on names, date of birth, date of death, state, country, spouse, etc with auto-complete capability.
* Estimated mandays: 7 
* Estimated number of worker: 1 (developer)
* Funding required: $2,632
* Estimated date of completion: ~~July 31~~ Sept 30, 2022

**Milestone: Automated Data Storage to Local IPFS**
* Currently, the process of storing our biographical data to our IPFS node is manual. Since storing of this data takes considerable amount of time, we aim to automate this process.
* We shall build a software module that could automatically retrieve biography data from our server and upload them to nft.storage (or equivalent) and then storing them in our local IPFS node. 
* Estimated mandays: 10 
* Estimated number of worker: 1 (developer)
* Funding required: $3,760 
* Estimated date of completion: ~~July 31~~ Sept 30, 2022

**Milestone: Optimizing Current Biography Publication and Storage System**
* The current publication and data storing process involves these steps - retrieving obituaries from external funeral sites automatically, manually converting the obituaries into biographies and manually uploading the biographies to IPFS and finally publishing the content on our site.
* It takes ~~up to~~ more than 2 minutes to complete one biography due to the manual processes involved.
* This milestone's aim is to automate some of the manual processes such as automatically upload content to IPFS and enabling some biographies to be published and uploaded completely autonomously. This will reduce human labor considerably thus allowing our platfom to scale. 
* This task requires the modification of our existing software and development of new modules.
* Estimated mandays: 30 
* Estimated number of worker: 1 (developer)
* Funding required: $11,280 ( ($47 per hour * 8 hours ) * 30 mandays )
* Estimated date of completion: ~~August 31~~ Oct 31, 2022

**Milestone: Publicity Campaign**
* At the moment, we have yet to officially launch our platform and understand there's a compelling reason to do so.
* Therefore, we need to begin our publicity effort via press releases, news/article coverage, online workshops with funeral homes and governmental agencies, blog posts, YouTube video production and more.
* We also aim to reach out to more funeral homes globally to encourage them to use our platform.
* We're more than happy to include Filecoin in all our activities.
* Estimated mandays: 40 
* Estimated number of worker: 2 (evangelist and content creator)
* Funding required: $15,040
* Estimated date of completion: ~~October 31~~ Dec 31, 2023

**Milestone: Translation to Four Other Languages**
* Since our mission is to be a globalized encyclopedia of the deceased, we wish to localize our platform to include four other major languages - Spanish, French, Hindi and Chinese.
* Estimated mandays: 20 
* Estimated number of worker: 4 (translator)
* Funding required: $7,520
* Estimated date of completion: ~~October 31~~ Dec 31, 2022

**Milestone: Object Data Storage Implementation**
* To implement object data storage capability on our system. We aim to use Seagate's CORTX object storage solution but the choice of vendor may change depending on cost-effectiveness.
* Compared to file or block file storage architectures, object storage is far better suited for large amounts of ever-growing media files that we use and offer better management at large scale and lower cost of storage.
* This milestone entails setting up our cloud server to support object storage and modifying our current software application to store and retrieve biographical data to and from an object storage facility.
* Estimated mandays: 20
* Estimated number of worker: 2 (developer and server administrator)
* Funding required: $7,520
* Estimated date of completion: November 30, 2022

**Milestone: Publish 500,000 Biographies**
* As of now, we've published over 50,000 biographical data and growing but at the most recent cost (May 15 to May 31) of ~~$0.15~~ $0.30 per biography. We ~~will run~~ already ran out of our microgrant funds to pay for publication ~~by end of May, 2022~~.
* From July, 2022 to January, 2023 (assuming this grant is approved), we hope to publish 500,000 biographies at only ~~$0.09~~ $0.20 each or lower.
* As comparison, it costs a mind-blowing $527 to publish an article on Wikipedia. Its 2021 total expenses amount to almost [$112 million](https://upload.wikimedia.org/wikipedia/foundation/1/1e/Wikimedia_Foundation_FY2020-2021_Audit_Report.pdf) while it publishes [582 articles in a day](https://en.wikipedia.org/wiki/Wikipedia:Statistics) on average, which gives us $527 per article. 
* Estimated mandays:  214 days
* Estimated number of worker: 10 (content editor)
* Funding required: ~~$45,000~~ $100,000 ($0.20 * 500,000)
* Estimated date of completion: ~~January 31~~ Mac 31, 2023

**~~Milestone: Host Our Own Filecoin Node~~**
* ~~To persistently backup all our biographical data in our own Filecoin node.~~
* ~~This milestone shall involve setting up a separate server and the necessary software to operate the node such as Lotus.~~
* ~~Estimated mandays: 7~~
* ~~Estimated number of worker: 1 (server administrator)~~
* ~~Funding required: $2,632~~
* ~~Estimated date of completion: January 31, 2023~~

**~~Milestone: Data Archiving Module~~**
* ~~This milestone involves developing a data migration and archiving module that allows us to efficiently collect and publish death records from other non-digital formats such as paper and digital formats such as Oracle, DB2, SQL, txt, json, xml, doc, pdf, bmp, gif, jpg, png and more.~~
* ~~This module shall be useful when we assist governmental departments to archive their death records, read milestone below.~~
* ~~It shall be able to map data structure from other formats to our format.~~
* ~~It also should be able to optically recognize text and images on image formats (such as after scanning).~~
* ~~This application would be substantially beneficial to other projects that are interested to archive different digital and non-digital data formats and utilize Filecoin for data storage, archival and publication.~~
* ~~Estimated mandays: 40~~
* ~~Estimated number of worker: 2 (developer)~~
* ~~Funding required: $15,040~~
* ~~Estimated date of completion: January 31, 2023~~

**~~Milestone: Governmental Outreach~~**
* ~~One of our plans is to assist governmental agencies to permanently and persistently archive their death records.~~
* ~~We aim to work with at least one governmental department to collect and store records of their deceased citizens on our platform.~~
* ~~This milestone requires the data archiving module in the milestone above.~~
* ~~We need to purchase 2 units of optical scanners such as [Fujitsu ScanSnap]( https://www.amazon.com/Fujitsu-ScanSnap-Versatile-Enabled-Document/dp/B08PH5Q51P/) at a cost of $500 each and two laptops to work along with these two scanners. These devices shall be used at government's premises to scan and digitize paper death records to be published on our platform. The cost for the scanners and laptops would be $2,000 ( ($500 * 2 scanners) + ($500 * 2 laptops) ).~~
* ~~Estimated mandays: 90~~
* ~~Estimated number of worker: 4 (advocate and content curator)~~
* ~~Funding required: $35,840 (manday cost + hardware cost)~~
* ~~Estimated date of completion: March 31, 2023~~

**Milestone: Server Hosting**
* We need to increase our cloud hosting capacity as our platform scales. We're currently using a single t2.medium of AWS EC2 instance.
* We need to upgrade to two units of t2.xlarge EC2 instances with 250GB of SDD storage. 
* Estimated monthly fee: ~~$250~~ $200 per month 
* Period of use: ~~June~~ Aug 2022 to ~~June~~ Aug 2023
* Funding required: ~~$3,000~~ $2,400

**Milestone: Local IPFS Node Server Hosting**
* We need to increase the capacity of our own local IPFS node. We're currently using a single t2.small of AWS EC2 instance.
* We need to upgrade to one unit of t2.large EC2 instances with 250GB of SDD storage. 
* Estimated monthly fee: $100 per month 
* Period of use: ~~June~~ Aug 2022 to ~~June~~ Aug 2023
* Funding required: $1,200

**~~Milestone: Local Filecoin Node Server Hosting~~**
* ~~To run our own node via one unit of t2.large EC2 instances with 250GB of SDD storage.~~
* ~~Estimated monthly fee: $100 per month~~
* ~~Period of use: June 2022 to June 2023~~
* ~~Funding required: $1,200~~

**Milestone: API Service**
* We currently relying on external API from APIFY to retrieve data from external sites.
* Estimated monthly fee: $150 per month 
* Period of use: ~~June~~ Aug 2022 to ~~June~~ Aug 2023
* Funding required: $1,800

**Milestone: Platform Administrator**
* To manage regular content curation and general platform administration.
* To maintain and keep our platform on a tip-top condition.
* We're estimated that this work requires an average of 2 hours per day.
* Estimated monthly fee: $2,820 per month 
* Period of use: ~~June~~ Aug 2022 to ~~June~~ Aug 2023
* Funding required: $33,840

## Total Budget Requested

Total requested budget is ~~$189,936~~ ~~$244,336~~ $189,624. The budget breakdown is detailed in the milestones above.

## Maintenance and Upgrade Plans

* Our ultimate mission is to document every single individual who have passed away in this world. Thus, we need the ability to digitally retrieve and store data in myriads of formats including in physical forms such as paper and microfilm (microfiche). This means it is in our best interest to upgrade our software in such a way that helps us achieve our goal.

* We shall also be looking at different long-term data storage formats to store our platform's biographies beyond our lifetime.

# Team

## Team Members
(NOTE: We aim to recruit more members as the need arises)

* Melvin Wong - Founder, Developer, Advocate
* Pedro Machado - Chief System Architect
* Shin Gahee - Advocate, Content Creator
* Prakash Reddy - Developer

## Team Member LinkedIn Profiles

* https://www.linkedin.com/in/melvinwong/
* https://www.linkedin.com/in/machadopedro/
* https://www.linkedin.com/in/shin-gahee-18882117/
* https://www.linkedin.com/in/prakashnalabolu/

## Team Website

https://lifefram.org

## Relevant Experience

Our team is the same team that won the grand prize of Babylon 2.0 blockchain hackathon in November 2021 sponsored by Filecoin. Subsequently to this, we received your generous microgrant and completed our milestones 20 days ahead of schedule.

We have utilized the funds from our hackathon prize, microgrant and own personal savings to turn this project from idea to reality.

As of now, more than 50,000 biographies have been published and we're adding around 1,000 new bios on a daily basis. We truly believe in the purpose of this project and hope Filecoin and other donors could support us. 

## Team code repositories

https://github.com/lifefram

# Additional Information

**Comment by melwong** [comment]
I'm so sorry. I missed out on a crucial milestone where we need a tech person as our platform administrator. I've added this in.

**Comment by melwong** [comment]
I'm so sorry @ErinOCon. I have miscalculated our most important milestone - to publish 500,000 biographies. Due to the recent inflation wave that's happening globally, our cost of publication has increased for the past 2 weeks. We're currently paying about $0.30 per biography, instead of $0.15 as previously stated.

Thus, I'd like to adjust our estimated future cost from USD0.09 to USD0.20 per bio. This is a more realistic forecast once we are able to scale our rate of publication.

Moreover, I also corrected the cost for the Server Hosting milestone from $250 per month to $200 per month.
